tests/sgl/device/test_device.cpp

```
@@ -9,6 +10,41 @@ using namespace sgl;
+{
+    SGL_UNUSED(user_object);
+    CHECK(user_data_size == sizeof(ExecuteCallbackTestState*));
+
+    auto state = *static_cast<ExecuteCallbackTestState* const*>(user_data);
+    state->called = true;
+    state->callback_handle = NativeHandle(context->nativeHandle);
+}
+
+void check_execute_callback_native_handle(Device* device, NativeHandle callback_handle)
+{
+    if (device->type() == DeviceType::d3d12) {
```

Cppcheck (2.20.0)

[style] The function 'class_name' is never used.

(unusedFunction)